broker/utils.go

```diff
@@ -10,19 +10,18 @@ import (
 	composeapi "github.com/compose/gocomposeapi"
 )
 
+var errDeploymentNotFound = errors.New("Deployment not found")
+
 func findDeployment(c compose.Client, name string) (*composeapi.Deployment, error) {
```

Comment:
TBH, this code:
```
	deploymentMeta, err := findDeployment(b.Compose, instanceName)
	if err == errDeploymentNotFound {
		return binding, brokerapi.ErrInstanceDoesNotExist
	} else if err != nil {
		return binding, err
	}
```
does almost the same thing as this one inside `findDeployment`:
```
deployment, errs := c.GetDeploymentByName(name)
	if len(errs) > 0 {
		if strings.Contains(errs[0].Error(), "deployment not found") {
			return nil, errDeploymentNotFound
		}
		return nil, compose.SquashErrors(errs)
	}
```
I don't see high value in keeping this as separate function if you have to do almost same condition outside and it forces us to write a lot of code for tests. However, this is minor thing and I commented on this only as you did some re-factoring - I'm not going to insist on it.